### DISCUSSION TITLE
Multiple signups

### DISCUSSION THREAD

**Discussion by altim** [Web3Auth/Web3Auth#1062] [Help]
How can I prevent user from signing up 2 (or more) times?
Because every time user signs-up he creates new key, basically destroying the previous one.
Since we are using the keys for the wallet, we obviously do not want the user to be able to destroy the key.
So, how can we make sure that we only let the user signup (create the new key) once?

**Comment by BboyStatix** [comment]
@altim what do you mean he creates a new key? Logging in with the account should always return the same key.

**Reply by ihor-repela** [reply]
@BboyStatix 
Lets say we have "signup" page where  we have Signup button which internally calls tKey._initializeNewKey()
And we have "login" page where we call just tKey.initialize()

Is there a way to check if user already has generated private key? Cause in other way he can go multiple times to Sign Up page and generate as many private keys as he wants (old keys will become invalid as I understand)
